src/MCPClient/tests/test_verify_checksum.py

```diff
@@ -0,0 +1,285 @@
+        assert job.get_stderr().decode("utf8").strip() == except_string_improper_format
+        mock.assert_called_once_with("-c", "--strict", hash_file)
+        assert ret == 1
+
+    def test_line_comparison_fail(self, mocker):
+        """If the checksum line and object comparison function fails then
+        we want to return early and _call shouldn't be called.
+        """
+        hash_file = "metadata/checksum.sha1"
+        hashsum = self.setup_hashsum(hash_file, Job("stub", "stub", ["", ""]))
+        toolname = "sha1sum"
+        mock = mocker.patch.object(hashsum, "_call", return_value=None)
+        mocker.patch.object(hashsum, "count_and_compare_lines", return_value=False)
+        mock.side_effect = subprocess.CalledProcessError(
+            returncode=1, cmd=toolname, output=None
+        )
+        ret = hashsum.compare_hashes(hash_file, "")
+        mock.assert_not_called()
+        assert ret == 1
+
+    @pytest.mark.parametrize(
+        "fixture",
+        [
+            ("checksum.md5", "md5"),
+            ("checksum.sha1", "sha1"),
+            ("checksum.sha256", "sha256"),
+            ("checksum_md5", "checksum_md5"),
+            ("checksum_sha1", "checksum_sha1"),
+            ("checksum_sha256", "checksum_sha256"),
+        ],
+    )
+    def test_get_ext(self, fixture):
+        """get_ext helps to format usefully."""
+        assert Hashsum.get_ext(fixture[0]) == fixture[1]
+
+    @staticmethod
+    def test_decode_and_version_string():
+        """Test that we can separate the version and license information
+        correctly from {command} --version.
+        """
+        version_string = (
+            b"sha256sum (GNU coreutils) 8.28\n"
+            b"Copyright (C) 2017 Free Software Foundation, Inc."
+        )
+        assert len(Hashsum._decode(version_string)[0]) > 1
+        assert Hashsum._decode(version_string)[0] == "sha256sum (GNU coreutils) 8.28"
+
+    @staticmethod
+    @pytest.fixture(scope="class")
+    def django_db_setup(django_db_blocker):
+        """Load the various database fixtures required for our tests."""
+        fixture_files = ["transfer.json", "files-transfer-unicode.json"]
+        fixtures = []
+        for fixture in fixture_files:
+            fixtures.append(os.path.join(THIS_DIR, "fixtures", fixture))
+        with django_db_blocker.unblock():
+            for fixture in fixtures:
+                call_command("loaddata", fixture)
+
+    @pytest.mark.django_db
+    def test_write_premis_event_to_db(self):
+        """Test that the microservice job connects to the database as
+        anticipated, writes its data, and that data can then be retrieved.
+        """
+        # Values the job writes.
+        group_type = "transfer_id"
+        event_type = "fixity_check"
+        event_outcome = "pass"
+        # Values we will write.
+        package_uuid = "e95ab50f-9c84-45d5-a3ca-1b0b3f58d9b6"
+        event_detail = "suma de verificación validada: OK"
```

Comment:
What's the black magic that make this localized string possible?!?! :)

Reply:
Ahh! None! You're right. But I want to keep on top of Unicode strings wherever possible. Who knows, we might write this out one day to the event details! :nerd_face: 

(But yep, really just pushing the database functions a little here so hope that's okay)